components/retro-go/drivers/audio/i2s.c

```diff
@@ -71,6 +71,9 @@ static bool driver_init(int device, int sample_rate)
                 .ws_io_num = RG_GPIO_SND_I2S_WS,
                 .data_out_num = RG_GPIO_SND_I2S_DATA,
                 .data_in_num = GPIO_NUM_NC
+            #if ESP_IDF_VERSION >= ESP_IDF_VERSION_VAL(4, 4, 0) // mck_io_num isn't present in esp-idf <= 4.3
```

Comment:
Can you move that to the top of the struct? so that it's the same order as defined by `i2s_pin_config_t`.